### ISSUE TITLE
Hearing Disposition - No Show

### ISSUE THREAD

**Issue by ShellySmithDSVA** [department-of-veterans-affairs/caseflow#8657]
## User Stories
As a Hearing Schedule user, I need to be able to record No Shows to hearings, put them on hold, and track the further actions needed to either cancel or reschedule the vets.

## Mocks
https://va-gov.invisionapp.com/share/7FPSIP28KD6#/screens
Disposition confirmation screen: https://va-gov.invisionapp.com/share/BCQH822EWS9#/346408072_Daily_Docket_Change_Alert

## Acceptance Criteria
- [x] Verify when No Show is selected, it automatically is assigned a "No Show waiting period" task
- [x] Verify when No Show is selected, the task stays assigned to the user the hearing was assigned to, but is automatically put on hold for 25 days and shows as on hold in that individual's queue
- [x] Verify the following task instructions are automatically added to the task "Mail must be received within 14 days of the original hearing date."
- [x] Verify the task can be resolved by choosing Mark as complete: Release case for review. This will close the task and mark the task as complete in both the Individual and Hearing Schedule team queues. 
- [x] Verify a confirmation modal with a descriptive message is presented when the user chooses Mark as complete: Release case for review
- [x] Verify the task can be resolved by choosing Mark as complete: Postponed. Choosing this option will change the disposition of the hearing to Postponed and present the user with the Postponement module of options. 
- [x] Verify the No Show task holds the case for 25 days
- [x] Verify the task has the option to be put on hold and a "Place on hold" option is available. Verify this blocks the No Show waiting period task 
- [x] Verify the Daily Docket is frozen as open the day of the hearing. Hearing dispositions can be changed throughout the hearing day. 
- [x] Verify that when a user changed a disposition and hits save, they are alerted with the confirmation modal.
- [x] Verify that a nightly job runs at midnight the day after the hearing that commits all changed dispositions. This allows the Daily Docket to be edited the day of the hearing and the following day.
- [x] Verify the Daily Docket is switched to read only after the nightly job is run and all dispositions are locked. 
- [x] Verify that if the hearing is marked as No Show and doesn’t change to postponement, the appeal is assigned an IHP task

LEGACY - LOGIC
- [x] Verify that legacy cases don't move from VACOLS location 38 (Awaiting Video Hearing) and VACOLS Location 36 (Awaiting CO Hearing) when a no-show task has been added. 
- [x] Verify that when no-show task is marked as "mark as complete: released case for review" then legacy cases move from VACOLS location 38 (Awaiting Video Hearing) and VACOLS Location 36 (Awaiting CO Hearing) to VACOLS Location 81 (Case Storage). If co-located VSO or DAV, then Legacy cases move to VACOLS location 55 (Veteran Service Organization).
- [x] ~~Verify that when no-show task is marked as "mark as complete and postpone," Then the legacy cases move from VACOLS location 38/36 to VACOLS location 57.~~ N/A 

LEGACY and AMA - LOGIC
- [x] Verify that if hearing disposition is changed to no show from another disposition, that previous tasks related to other dispositions are removed (Can they be deleted?) and no show moves AMA flow forward, and moves Legacy cases as needed.

## Related Tickets
See Epic #8650 for the complete scope of work related to this ticket.
See #8655 for the full Postponement disposition flow.

**Comment by lomaxap** [comment]
- [ ] Verify when No Show is selected, the task stays assigned to the user the hearing was assigned to

should [the no show task](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/tasks/disposition_task.rb#L97) be assigned to an actual user?
- [ ] Verify the following task instructions are automatically added to the task "Mail must be received within 14 days of the original hearing date."

also need to add default instructions [here](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/tasks/disposition_task.rb#L102)
- [ ] Verify that if the hearing is marked as No Show and doesn’t change to postponement, the appeal is assigned an IHP task

unclear to me if this is happening. Should confirm that [`create_ihp_tasks!`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/tasks/hearing_task.rb#L37) is called after on hold period

**Comment by tomas-nava** [comment]
> _Verify when No Show is selected, the task stays assigned to the user the hearing was assigned to, but is automatically put on hold for 25 days and shows as on hold in that individual's queue_

> should the no show task be assigned to an actual user?

The way I read this, the `DispositionTask` (or `ChangeHearingDispositionTask`) is the task that "stays assigned to the user the hearing was assigned to". This is the current functionality; you can test by adding a `binding.pry` after [the last line of this test](https://github.com/department-of-veterans-affairs/caseflow/blob/master/spec/feature/hearing_schedule/change_hearing_disposition_spec.rb#L201), then changing the hearing disposition to No Show. The `ChangeHearingDispositionTask` will appear in the user's On Hold tab with a 25 day hold.

> _Verify that if the hearing is marked as No Show and doesn’t change to postponement, the appeal is assigned an IHP task_

> unclear to me if this is happening.

Yeah, I'm pretty sure this isn't happening yet.

**Comment by sharonwarner** [comment]
The IHP tasks and VACOLS location logic runs by default from the hearing task when its children tasks are closed. I submitted a PR above to add the instructions to the created no show hearing task.
